python/examples/llama/test_llama3.py

```diff
@@ -0,0 +1,1861 @@
+# RUN: %pytest %s
+
+from dataclasses import dataclass
+import functools
+import math as pymath
+import pytest
+import torch
+from typing import Optional, Tuple
+
+from mlir import ir
+from mlir.dialects import transform, func, linalg, tensor, arith, complex, math
+from mlir.dialects.transform import structured
+from mlir.dialects.transform import interpreter
+from mlir.passmanager import PassManager
+from mlir.runtime.np_to_memref import (
+    get_ranked_memref_descriptor,
+)
+from mlir.execution_engine import ExecutionEngine
+
+
+from lighthouse import utils as lh_utils
+
+
+def with_mlir_ctx(ctx: ir.Context):
+    def with_mlir_ctx_decorator(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            with ctx, ir.Location.unknown(context=ctx):
+                return func(*args, **kwargs)
+
+        return wrapper
+
+    return with_mlir_ctx_decorator
+
+
+@dataclass
+class ModelArgs:
+    dim: int = 4096
+    n_layers: int = 32
+    n_heads: int = 32
+    n_kv_heads: Optional[int] = None
+    vocab_size: int = -1
+    multiple_of: int = 256  # make SwiGLU hidden layer size multiple of large power of 2
+    ffn_dim_multiplier: Optional[float] = None
+    norm_eps: float = 1e-5
+    rope_theta: float = 500000
+
+    max_batch_size: int = 32
+    max_seq_len: int = 2048
+
+
+def affine_map(dim_count, exprs, *, symb_count=0):
+    return ir.AffineMap.get(dim_count, symb_count, exprs)
+
+
+parallel = linalg.IteratorType.parallel
+reduction = linalg.IteratorType.reduction
+
+
+def create_pass_pipeline(ctx: ir.Context) -> PassManager:
+    with ctx:
+        pm = PassManager("builtin.module")
+        pm.add("convert-scf-to-cf")
+        pm.add("expand-strided-metadata")
+        pm.add("lower-affine")
+        pm.add("finalize-memref-to-llvm")
+        pm.add("convert-func-to-llvm")
+        pm.add("convert-to-llvm")
+        pm.add("reconcile-unrealized-casts")
+        pm.add("cse")
+        pm.add("canonicalize")
+    return pm
+
+
+def create_schedule(ctx: ir.Context) -> ir.Module:
+    """
+    Create an MLIR module containing transformation schedule.
+    The schedule provides partial lowering to scalar operations.
+
+    Args:
+        ctx: MLIR context.
+    """
+    with ctx, ir.Location.unknown(context=ctx):
+        # Create transform module.
+        schedule = ir.Module.create()
```

Comment:
```suggestion
def create_schedule() -> ir.Module:
    schedule = ir.Module.create()
```
And just de-indent the rest of the function.